Assets/Scripts/GameManagement/GameManager.cs

```diff
@@ -1,32 +1,68 @@
-﻿using UnityEngine;
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
 using UnityEngine.SceneManagement;
+using Moonshot.UI;
 
 namespace Moonshot.GameManagement {
 
 	public class GameManager : MonoBehaviour {
 		#region Singleton
-		private static GameManager instance = null;
-		private static readonly object padlock = new object();
+		public static GameManager Instance { get; private set; }
 
-		GameManager() { }
-
-		public static GameManager Instance {
-			get {
-				lock (padlock) {
-					if (instance == null) {
-						instance = new GameManager();
-					}
-					return instance;
-				}
+		void Awake() {
+			if (Instance == null) {
+				Instance = this;
+				DontDestroyOnLoad(gameObject);
+			} else {
+				DestroyImmediate(gameObject);
+				return;
 			}
 		}
 		#endregion
 
+		[SerializeField] private GameObject LoadingScreen;
+		[SerializeField] private ProgressBar m_ProgressBar;
+
+		List<AsyncOperation> scenesLoading = new List<AsyncOperation>();
+		float TotalProgress;
+
 		#region Unity Methods
-		private void Awake() {
-			SceneManager.LoadSceneAsync((int)SceneIndexes.TITLE_SCENE, LoadSceneMode.Additive);
+		private void Start() {
+			SceneManager.LoadSceneAsync((int)SceneIndexes.TITLE_SCREEN, LoadSceneMode.Additive);
+		}
+		#endregion
+
+		#region Custom Methods
+		public void LoadGame() {
+			LoadingScreen.SetActive(true);
+
+			scenesLoading.Add(SceneManager.UnloadSceneAsync((int)SceneIndexes.TITLE_SCREEN));
+			scenesLoading.Add(SceneManager.LoadSceneAsync((int)SceneIndexes.MAIN, LoadSceneMode.Additive));
+
+			StartCoroutine(GetSceneLoadProgress());
```

Comment:
Could we rename this method? I'm getting an "ambiguous method call" error since the method is being overloaded with the same function signature.
![image](https://user-images.githubusercontent.com/11526734/98060886-49b76d80-1e10-11eb-87cf-33a98ee0b04e.png)

Reply:
That looks like some kind of import/compilation error... it's ambiguous between the same two methods?

Reply:
Fixed...